#! /bin/sh /usr/share/dpatch/dpatch-run
## lmce_keybindings.dpatch by Daniel Kristjansson <danielk@cuymedia.net>

@DPATCH@
diff -urNad mythtv-0.20+fixes15192~/libs/libmythtv/dbcheck.cpp mythtv-0.20+fixes15192/libs/libmythtv/dbcheck.cpp
--- blah/libs/libmythtv/dbcheck.cpp	(revision 15192)
+++ blah/libs/libmythtv/dbcheck.cpp	(working copy)
@@ -439,6 +439,12 @@
     query.exec();
     /// Remove any old TMS labs grabber. -- end
 
+    /// Remove any LMCE conflicting keybindings -- begin
+    query.exec("UPDATE keybindings SET keylist='' WHERE keylist='F6'");
+    query.exec("UPDATE keybindings SET keylist='' WHERE keylist='F7'");
+    query.exec("UPDATE keybindings SET keylist='' WHERE keylist='F8'");
+    /// Remove any LMCE conflicting keybindings -- end
+
     if (dbver == currentDatabaseVersion)
         return true;
 
diff -urNad mythtv-0.20+fixes15192~/libs/libmythtv/tv_play.cpp mythtv-0.20+fixes15192/libs/libmythtv/tv_play.cpp
--- blah/libs/libmythtv/tv_play.cpp	(revision 15192)
+++ blah/libs/libmythtv/tv_play.cpp	(working copy)
@@ -184,7 +184,7 @@
     REG_KEY("TV Playback", "PLAY", "Play", "Ctrl+P");
     REG_KEY("TV Playback", "JUMPPREV", "Jump to previously played recording", "");
     REG_KEY("TV Playback", "JUMPREC", "Display menu of recorded programs to jump to", "");
-    REG_KEY("TV Playback", "SIGNALMON", "Monitor Signal Quality", "F7");
+    REG_KEY("TV Playback", "SIGNALMON", "Monitor Signal Quality", "");
     REG_KEY("TV Playback", "JUMPTODVDROOTMENU", "Jump to the DVD Root Menu", "");
 
     /* Editing keys */
@@ -210,8 +210,8 @@
     REG_KEY("Teletext Menu", "MENUGREEN",   "Menu Green",            "F3");
     REG_KEY("Teletext Menu", "MENUYELLOW",  "Menu Yellow",           "F4");
     REG_KEY("Teletext Menu", "MENUBLUE",    "Menu Blue",             "F5");
-    REG_KEY("Teletext Menu", "MENUWHITE",   "Menu White",            "F6");
-    REG_KEY("Teletext Menu", "TOGGLEBACKGROUND","Toggle Background", "F7");
+    REG_KEY("Teletext Menu", "MENUWHITE",   "Menu White",            "");
+    REG_KEY("Teletext Menu", "TOGGLEBACKGROUND","Toggle Background", "");
     REG_KEY("Teletext Menu", "REVEAL",      "Reveal hidden Text",    "F8");
 
     /* Interactive Television keys */
@@ -219,8 +219,8 @@
     REG_KEY("ITV Menu", "MENUGREEN",  "Menu Green",  "F3");
     REG_KEY("ITV Menu", "MENUYELLOW", "Menu Yellow", "F4");
     REG_KEY("ITV Menu", "MENUBLUE",   "Menu Blue",   "F5");
-    REG_KEY("ITV Menu", "TEXTEXIT",   "Menu Exit",   "F6");
-    REG_KEY("ITV Menu", "MENUTEXT",   "Menu Text",   "F7");
+    REG_KEY("ITV Menu", "TEXTEXIT",   "Menu Exit",   "");
+    REG_KEY("ITV Menu", "MENUTEXT",   "Menu Text",   "");
     REG_KEY("ITV Menu", "MENUEPG",    "Menu EPG",    "F12");
 /*
   keys already used:
 
